### PR TITLE
linux_6_5: init, stable/LTS kernels upgrade

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

https://lwn.net/Articles/942879/

Here we go, fast boots on 256-cores systems!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by K900 — APPROVED** [review]
Diff LGTM, going to test on my laptop.

File: pkgs/os-specific/linux/kernel/linux-6.5.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  # modDirVersion needs to be x.y.z, will automatically add .0 if needed
+  modDirVersion = versions.pad 3 version;
+
+  # branchVersion needs to be x.y
+  extraMeta.branch = versions.majorMinor version;
+
+  src = fetchurl {
+    url = "mirror://kernel/linux/kernel/v6.x/linux-${version}.tar.xz";
+    hash = lib.fakeHash;
```
Comment (K900) [inline pkgs/os-specific/linux/kernel/linux-6.5.nix:16]: Missed this

**Comment by K900** [comment]
Run-tested on NixOS test VM, Hetzner VM, RPi4, Asus GA402 laptop.

**Comment by RaitoBezarius** [comment]
Hardened stuff dropped for 6.5.

**Comment by RaitoBezarius** [comment]
(Feel free to merge, I will be over the air for the next hours.)

**Comment by vcunat** [comment]
This broke 5.10 with
```
../sound/soc/intel/boards/sof_sdw.c:208:41: error: 'RT711_JD2_100K' undeclared here (not in a function)
  208 |                 .driver_data = (void *)(RT711_JD2_100K),
      |                                         ^~~~~~~~~~~~~~
```
Hydra e.g. https://hydra.nixos.org/build/233264524

**Comment by Ma27** [comment]
5.10.193 seems fine again locally.

**Pull request merged**